ui/src/store/useSituationsStore.ts

```diff
@@ -1,28 +1,65 @@
 import { defineStore } from 'pinia'
-import { getSituations } from '@/services/AlecService'
-import { TSituation } from '@/types/TSituation'
-import { groupBy } from 'lodash'
+import {
+	getAllAlarms,
+	getSituations,
+	getAllNodes
+} from '@/services/AlarmService'
+import { getSituationsStatus } from '@/services/AlecService'
+
+import { TAlarm, TNode, TSituation, TSituationSaved } from '@/types/TSituation'
+import { groupBy, mapKeys } from 'lodash'
 
 type TState = {
 	situations: TSituation[]
+	totalSituations: number
+	nodes: TNode[]
 }
 
 export const useSituationsStore = defineStore('situationsStore', {
 	state: (): TState => ({
-		situations: []
+		situations: [],
+		totalSituations: 0,
+		nodes: []
 	}),
 	actions: {
+		async getNodes() {
+			this.nodes = []
+			const result = await getAllNodes()
+			if (result) {
+				this.nodes = result
+			}
+		},
 		async getSituations() {
 			this.situations = []
-			const situations = await getSituations()
-			if (situations) {
+			const result = await getSituations()
+			const resultAlarms = await getAllAlarms()
+			let allAlarms: Record<number, TAlarm> = []
+			if (resultAlarms) {
+				allAlarms = mapKeys(resultAlarms, (a) => a.id)
+			}
+
+			const resultStatus = await getSituationsStatus()
+			this.totalSituations = result.totalCount
+			if (result) {
+				const situations = result.alarm.map((s: TSituation) => {
+					const alarms = s.relatedAlarms.map(
+						(a: TAlarm) => allAlarms[parseInt(a.id)]
```

Comment:
Not sure exactly what these ids consist of, but is it possible it could default to radix 16? Usually I do `parseInt(a.id, 10)` just to be safe. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt